server/util/csv_parse.py

```diff
@@ -258,14 +261,37 @@ def reject_duplicate_values(
         yield row
 
 
+TOTAL_REGEX = re.compile(r"(^|[^a-zA-Z])(sub)?totals?($|[^a-zA-Z])", re.IGNORECASE)
+
+
 def reject_total_rows(csv: CSVDictIterator) -> CSVDictIterator:
     for r, row in enumerate(csv):  # pylint: disable=invalid-name
         for value in row.values():
-            if value.lower() in ["total", "totals", "total ballots", "county totals"]:
-                raise CSVParseError(f"Remove total row (row {r+2})")
+            if TOTAL_REGEX.search(value):
+                raise CSVParseError(
+                    f"It looks like you might have a total row (row {r+2})."
+                    " Please remove this row from the CSV."
+                )
+        yield row
 
+
+def reject_final_total_row(csv: CSVDictIterator, columns: List[CSVColumnType]):
+    column_values = defaultdict(list)
+
+    for row in csv:
+        for column in columns:
+            value = row.get(column.name)
+            if column.value_type == CSVValueType.NUMBER and value is not None:
+                column_values[column.name].append(value)
         yield row
 
+    for values in column_values.values():
+        if sum(values[:-1]) == values[-1]:
```

Comment:
Is there a chance for this to be a false positive?

Reply:
For sure, but I guess that's a risk we're willing to take? If it's a false positive, a possible fix is swapping that row with a different row (since it only checks the last row)